File: cpp/3101.count-alternating-subarrays.cpp
#include <vector>
using std::vector;

class Solution {
public:
  long long countAlternatingSubarrays(vector<int> &nums) {
    long total = 0L;

    int cur = -1, curLen = 0;

    for (int num : nums) {
      if (num == cur) {
        curLen = 1;
      } else {
        curLen++;
      }

      total += curLen;
      cur = num;
    }

    return total;
  }
};

File: golang/3101.count-alternating-subarrays.go
func countAlternatingSubarrays(nums []int) int64 {
	total := 0
	cur, curLen := -1, 0

	for _, num := range nums {

		if num == cur {
			curLen = 1
		} else {
			curLen++
		}

		total += curLen
		cur = num
	}

	return int64(total)
}
